src/WindowManager.cc

```diff
@@ -1475,7 +1475,8 @@ WindowManager::createClient(Window window, bool is_new)
 
                 if (wo != nullptr
                     && wo->isMapped()
-                    && (wo->getParent()->isFullscreen()
+                    && ((pekwm::config()->isFullscreenAbove()
+                         && wo->getParent()->isFullscreen())
```

Comment:
This needs to be (and should have been in the last commit)
```
wo->getParent() && wo->getParent()->isFullscreen()
```
Question is why the it is not enough to do wo->isFullscreen() as that state should be propagated on the active client of a frame.

Reply:
I don't know. Maybe it's too early or something? I probably can add some tracing and find out why. But it'll take a while

Reply:
OK it's not propagated because `_fullscreen` is actually a member of PDecor (before I moved it to PWinObj) so Client object is oblivious about it. Client can change fullscreen state via EWMH hints, but I don't suppose toggling fullscreen changes that, not sure.

Client::isFullscreen is implemented to return `_state.fullscreen` which is separate from `this->_fullscreen`. worse `PWinObj::isFullscreen` is not a virtual method and `Client::isFullscreen` overrides it. This is kinda messy, maybe moving isFullscreen back to PWinObj was a wrong move.

Reply:
I actually think it sounds like a good idea to remove/not use _state.fullscreen and then use PWinObj fullscreen instead (that without looking at the code, there might be reasons why it becomes inconvenient)

But, I'll merge this for now if you're ok with it to avoid crashes on master.

Reply:
5 minute, i think i could avoid the getParent...

Reply:
Take your time! :)

Reply:
OK tested working. I still don't understand how Client::isFullscreen gets the info which is set by Frame::setStateFullscreen, but I can do that in my own time.